test/utils/parser_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/kr/pretty"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestParseK8sYAML(t *testing.T) {
+	headerText := "# Copyright 2023 Google LLC\n" +
+		"#\n" +
+		"# Licensed under the Apache License, Version 2.0 (the \"License\");" +
+		"# you may not use this file except in compliance with the License." +
+		"# You may obtain a copy of the License at" +
+		"#\n" +
+		"#   https://www.apache.org/licenses/LICENSE-2.0" +
+		"#\n" +
+		"# Unless required by applicable law or agreed to in writing, software" +
+		"# distributed under the License is distributed on an \"AS IS\" BASIS," +
+		"# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied." +
+		"# See the License for the specific language governing permissions and" +
+		"# limitations under the License." +
+		"\n"
+
+	ingText := "apiVersion: networking.k8s.io/v1\nkind: Ingress\nmetadata:\n  name: foo-internal\n  " +
```

Comment:
for large blocks like this, you should use golang raw string literals:
https://askgolang.com/golang-raw-string-literals/

`apiVersion: ...

Reply:
Updated. Thanks for letting me know!